xds/src/main/java/io/grpc/xds/RbacFilter.java

```diff
@@ -0,0 +1,308 @@
+import java.util.logging.Logger;
+import javax.annotation.Nullable;
+
+
+/** RBAC Http filter implementation. */
+final class RbacFilter implements Filter, ServerInterceptorBuilder {
+  private static final Logger logger = Logger.getLogger(RbacFilter.class.getName());
+
+  static final RbacFilter INSTANCE = new RbacFilter();
+
+  static final String TYPE_URL =
+          "type.googleapis.com/envoy.extensions.filters.http.rbac.v3.RBAC";
+
+  RbacFilter() {}
+
+  @Override
+  public String[] typeUrls() {
+    return new String[] { TYPE_URL };
+  }
+
+  @Override
+  public ConfigOrError<? extends FilterConfig> parseFilterConfig(Message rawProtoMessage) {
+    RBAC rbacProto;
+    if (!(rawProtoMessage instanceof Any)) {
+      return ConfigOrError.fromError("Invalid config type: " + rawProtoMessage.getClass());
+    }
+    Any anyMessage = (Any) rawProtoMessage;
+    try {
+      rbacProto = anyMessage.unpack(RBAC.class);
+    } catch (InvalidProtocolBufferException e) {
+      return ConfigOrError.fromError("Invalid proto: " + e);
+    }
+    return parseRbacConfig(rbacProto);
+  }
+
+  @VisibleForTesting
+  @Nullable
+  static ConfigOrError<RbacConfig> parseRbacConfig(RBAC rbac) {
+    // https://github.com/envoyproxy/envoy/blob/a12869fa9e9add4301a700978d5489e6a0cc0526/api/envoy/extensions/filters/http/rbac/v3/rbac.proto#L25
+    if (!rbac.hasRules()) {
+      return null;
+    }
+    io.envoyproxy.envoy.config.rbac.v3.RBAC rbacConfig = rbac.getRules();
+    GrpcAuthorizationEngine.Action authAction;
+    switch (rbacConfig.getAction()) {
+      case ALLOW:
+        authAction = GrpcAuthorizationEngine.Action.ALLOW;
+        break;
+      case DENY:
+        authAction = GrpcAuthorizationEngine.Action.DENY;
+        break;
+      case LOG:
+      default:
+        return null;
+    }
+    Map<String, Policy> policyMap = rbacConfig.getPoliciesMap();
+    List<GrpcAuthorizationEngine.PolicyMatcher> policyMatchers = new ArrayList<>();
+    for (Map.Entry<String, Policy> entry: policyMap.entrySet()) {
+      try {
+        Policy policy = entry.getValue();
+        if (policy.hasCondition() || policy.hasCheckedCondition()) {
+          return ConfigOrError.fromError(
+                  "Policy.condition and Policy.checked_condition must not set.");
+        }
+        policyMatchers.add(new PolicyMatcher(entry.getKey(),
+                parsePermissionList(policy.getPermissionsList()),
+                parsePrincipalList(policy.getPrincipalsList())));
+      } catch (Exception e) {
+        logger.log(Level.FINE, "Unexpected error parsing policy {0}", new Object[]{entry});
+        return ConfigOrError.fromError("Encountered error parsing policy: " + e);
+      }
+    }
+    return ConfigOrError.fromConfig(RbacConfig.create(new AuthConfig(policyMatchers, authAction)));
+  }
+
+  @Override
+  public ConfigOrError<? extends FilterConfig> parseFilterConfigOverride(Message rawProtoMessage) {
+    return parseFilterConfig(rawProtoMessage);
```

Comment:
This doesn't seem right. The override RBAC config is [`RBACPerRoute`](https://github.com/grpc/grpc-java/blob/91948b26061bb88164e7f5cc791b4ecf398655f3/xds/third_party/envoy/src/main/proto/envoy/extensions/filters/http/rbac/v3/rbac.proto#L39).

Reply:
I see. 
So if after parsing this if we see [rbac](https://github.com/grpc/grpc-java/blob/91948b26061bb88164e7f5cc791b4ecf398655f3/xds/third_party/envoy/src/main/proto/envoy/extensions/filters/http/rbac/v3/rbac.proto#L46) field being absent, then in the returned `RbacConfig` we need a way to indicate that the global rbacfilter needs to be disabled.  We might return the `RbacConfig` with `authConfig` being null. Then when returning from `buildServerInterceptor` we return null which means no interceptor should be installed. Similar to here https://github.com/grpc/grpc-java/blob/b7f3fddc76ecfe854c6039c000ff6d168205ab06/xds/src/main/java/io/grpc/xds/XdsNameResolver.java#L418

Reply:
Right, looks we would need such a mechanism to "revoke" the top-level RBAC config. Using the approach of nullable `authConfig` in `RbacConfig` sounds fine.